### PR TITLE
tests: Cleanup gofail

### PR DESCRIPTION
**Pull request by serathius**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.
cc @ahrtr

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,6 +105,27 @@ verify-proto-annotations:
...
+gofail-disable:
+	gofail disable server/etcdserver/ server/storage/backend/ server/storage/mvcc/
+	cd ./server && go mod tidy
+	cd ./etcdutl && go mod tidy
+	cd ./etcdctl && go mod tidy
+	cd ./tests && go mod tidy
+
+.PHONY: install-gofail
+install-gofail:
+	cd tools/mod; go install go.etcd.io/gofail@cb2ba30a3077
```
Comment (ahrtr) [inline Makefile:128]: Can you use the latest commit ID fc21f61ba88a639ebd199a33bbb298093b21b0ed?

Comment (serathius) [reply Makefile:128]: Sure

File: Makefile [hunk 2]
```diff
@@ -105,6 +105,29 @@ verify-proto-annotations:
...
+
+GOFAIL_VERSION=fc21f61ba88a639ebd199a33bbb298093b21b0ed
+
+.PHONY: gofail-enable
+gofail-enable: install-gofail
+	gofail enable server/etcdserver/ server/storage/backend/ server/storage/mvcc/
+	cd ./server && go get go.etcd.io/gofail/runtime@${GOFAIL_VERSION}
+	cd ./etcdutl && go get go.etcd.io/gofail/runtime@${GOFAIL_VERSION}
+	cd ./etcdctl && go get go.etcd.io/gofail/runtime@${GOFAIL_VERSION}
+	cd ./tests && go get go.etcd.io/gofail/runtime@${GOFAIL_VERSION}
```
Comment (ahrtr) [inline Makefile:118]:
```suggestion
	cd ./server && go get go.etcd.io/gofail@${GOFAIL_VERSION}
	cd ./etcdutl && go get go.etcd.io/gofail@${GOFAIL_VERSION}
	cd ./etcdctl && go get go.etcd.io/gofail@${GOFAIL_VERSION}
	cd ./tests && go get go.etcd.io/gofail@${GOFAIL_VERSION}
```

File: Makefile [hunk 3]
```diff
@@ -30,8 +30,8 @@ test-e2e-release: build
 	PASSES="release e2e" ./scripts/test.sh $(GO_TEST_FLAGS)
 
 .PHONY: test-linearizability
-test-linearizability:
-	FAILPOINTS=true PASSES="linearizability" ./scripts/test.sh $(GO_TEST_FLAGS)
+test-linearizability: gofail-enable
```
Comment (ahrtr) [inline Makefile:33]: It doesn't make sense to depend on `gofail-enable`.  It should only depend on binaries built with gofail enabled. 
```suggestion
test-linearizability: 
```
```suggestion
test-linearizability:
```

File: Makefile [hunk 4]
```diff
@@ -105,6 +105,29 @@ verify-proto-annotations:
...
+.PHONY: gofail-enable
+gofail-enable: install-gofail
+	gofail enable server/etcdserver/ server/storage/backend/ server/storage/mvcc/
+	cd ./server && go get go.etcd.io/gofail@${GOFAIL_VERSION}
+	cd ./etcdutl && go get go.etcd.io/gofail@${GOFAIL_VERSION}
+	cd ./etcdctl && go get go.etcd.io/gofail@${GOFAIL_VERSION}
+	cd ./tests && go get go.etcd.io/gofail@${GOFAIL_VERSION}
+
+.PHONY: gofail-disable
+gofail-disable:
```
Comment (ahrtr) [inline Makefile:121]:
```suggestion
gofail-disable: install-gofail
```

**Review by ahrtr — APPROVED** [review]
LGTM

**Comment by ahrtr** [comment]
Please also bump gofail in tools/mod as well.

**Comment by serathius** [comment]
Updated makefile to read the gofail version from go.mod file

File: Makefile [hunk 5]
```diff
@@ -105,6 +105,29 @@ verify-proto-annotations:
 verify-genproto:
 	PASSES="genproto" ./scripts/test.sh
 
+# Failpoints
+
+GOFAIL_VERSION = $(shell cd tools/mod && go mod graph | grep go.etcd.io/etcd/tools/v3 | grep go.etcd.io/gofail | cut -d'@' -f 2)
```
Comment (ahrtr) [inline Makefile:110]:
```suggestion
GOFAIL_VERSION = $(go list -m -f {{.Version}} go.etcd.io/gofail)
```

**Pull request merged**